### DISCUSSION TITLE
Feature ideas and possible use cases feedback

### DISCUSSION THREAD

**Discussion by ghsoares** [ghsoares/ReactGD#3] [General]
Hello everyone!

I'm really happy to know that this tool will be useful for you, so I've created this discussion to know which features are needed for your projects and for which use cases this tool could be used.

Right now I'm focused on runtime utility, this means all the features that are used while your project is being executed, for example:
- UI tree composition;
- Property changes;
- Signal handling;
- Component state handling;
- Event handling;
- Theme;
- Tween animations;
- And so on.

After completing all the necessary features, I'll re-implement GDX, the GDScript extension language used by ReactGD that includes special syntax to describe the rendered component nodes more easily.

For now, I want to know what features you are planning to use in your project and which features are missing, also if possible, which use case are you planning to use ReactGD, is it for a shop? Player inventory? Status HUD? It's important to let me know so I can optimize this tool for most used cases.

If you want to take a look at current features, go to [Prototype3 branch](https://github.com/ghsoares/ReactGD/tree/Prototype3), maybe this will be the last prototype branch before going into first actual version.

Thank you for your support!

**Comment by gabbydelforge** [comment]
As a game developer with a background in web dev and a passion for React, I'm very happy to learn about this repo. I'm working on a new Godot project that has quite a large UI component, so I was actually looking into creating something similar to this myself. I have just started playing around with it and it's promising so far.

At this point, here are the features I'm missing the most:
- Hot reloading (I've been spoiled by Create React App, but this really is a huge workflow boost)
- Autocomplete / IntelliSense for props (perhaps in VSCode?)
- An easy way to set Layout Presets for Components (ie. "Full Rect" - I use this one often)
